packages/sveltekit/src/vite/autoInstrument.ts

```diff
@@ -0,0 +1,123 @@
+  debug: boolean;
+};
+
+/**
+ * Creates a Vite plugin that automatically instruments the parts of the app
+ * specified in @param options. This includes
+ *
+ * - universal `load` functions from `+page.(js|ts)` and `+layout.(js|ts)` files
+ * - server-only `load` functions from `+page.server.(js|ts)` and `+layout.server.(js|ts)` files
+ *
+ * @returns the plugin
+ */
+export async function makeAutoInstrumentationPlugin(options: AutoInstrumentPluginOptions): Promise<Plugin> {
+  const { load: shouldWrapLoad, serverLoad: shouldWrapServerLoad, debug } = options;
+
+  return {
+    name: 'sentry-auto-instrumentation',
+    // This plugin needs to run as early as possible, before the SvelteKit plugin virtualizes all paths and ids
+    enforce: 'pre',
+
+    async load(id) {
+      const applyUniversalLoadWrapper =
+        shouldWrapLoad &&
+        /^\+(page|layout)\.(js|ts|mjs|mts)$/.test(path.basename(id)) &&
+        (await canWrapLoad(id, debug));
+
+      if (applyUniversalLoadWrapper) {
+        // eslint-disable-next-line no-console
+        debug && console.log(`Wrapping ${id} with Sentry load wrapper`);
+        return getWrapperCode('wrapLoadWithSentry', `${id}${WRAPPED_MODULE_SUFFIX}`);
+      }
+
+      const applyServerLoadWrapper =
+        shouldWrapServerLoad &&
+        /^\+(page|layout)\.server\.(js|ts|mjs|mts)$/.test(path.basename(id)) &&
+        (await canWrapLoad(id, debug));
+
+      if (applyServerLoadWrapper) {
+        // eslint-disable-next-line no-console
+        debug && console.log(`Wrapping ${id} with Sentry load wrapper`);
+        return getWrapperCode('wrapServerLoadWithSentry', `${id}${WRAPPED_MODULE_SUFFIX}`);
+      }
+
+      return null;
+    },
+  };
+}
+
+/**
+ * We only want to apply our wrapper to files that
+ *
+ *  - Have no Sentry code yet in them. This is to avoid double-wrapping or interferance with custom
+ *    Sentry calls.
+ *  - Actually declare a `load` function. The second check of course is not 100% accurate, but it's good enough.
+ *    Injecting our wrapper into files that don't declare a `load` function would result in a build-time warning
+ *    because vite/rollup warns if we reference an export from the user's file in our wrapping code that
+ *    doesn't exist.
+ *
+ * Exported for testing
+ *
+ * @returns `true` if we can wrap the given file, `false` otherwise
+ */
+export async function canWrapLoad(id: string, debug: boolean): Promise<boolean> {
+  const code = (await fs.promises.readFile(id, 'utf8')).toString();
+
+  const codeWithoutComments = code.replace(/(\/\/.*| ?\/\*[^]*?\*\/)(,?)$/gm, '');
+
+  const hasSentryContent = codeWithoutComments.includes('@sentry/sveltekit');
+  if (hasSentryContent) {
+    // eslint-disable-next-line no-console
+    debug && console.log(`Skipping wrapping ${id} because it already contains Sentry code`);
+  }
+
+  const hasLoadDeclaration = /(const|let|var|function)\s+load\s*(=|\()/gm.test(codeWithoutComments);
```

Comment:
I thought about this, too but I think this isn't valid syntax: https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#syntax (🤞 )

Reply:
Ah yeah sorry missed something in the snippet - you gotta use `as` for esm, but this still applies if I'm thinking about this correctly.

```ts
export {
  someThingElse as load
}
```

Reply:
Ahh damnit, I missed this case. I think we could adjust the regex to test for `/as\s+load/` but I guess this has the potential to interfere with TypeScript's type assertions. However, in the worst case, this would produce a build warning so I think it's fine.